src/testRunner/unittests/tsserverProjectSystem.ts

```diff
@@ -3493,6 +3493,48 @@ namespace ts.projectSystem {
             }
         });
 
+        function createSessionWithEventHandler(host: TestServerHost) {
+            const surveyEvents: server.SurveyReady[] = [];
+            const session = createSession(host, {
+                eventHandler: e => {
+                    if (e.eventName === server.SurveyReady) {
+                        surveyEvents.push(e);
+                    }
+                }
+            });
+
+            return { session, verifySurveyReadyEvent };
+
+            function verifySurveyReadyEvent() {
+                assert.equal(surveyEvents.length, 1);
+                assert.deepEqual(surveyEvents, [{
+                    eventName: server.SurveyReady,
+                    data: { surveyId: "checkJs" }
+                }]);
+            }
+        }
+
+        it("logs an event when checkJs is set", () => {
+            const projectRoot = "/user/username/projects/project";
+            const file: File = {
+                path: `${projectRoot}/src/file.ts`,
+                content: "export var y = 10;"
+            };
+            const tsconfig: File = {
+                path: `${projectRoot}/tsconfig.json`,
+                content: JSON.stringify({ compilerOptions: { checkJs: true } }),
+            };
+            const host = createServerHost([file, tsconfig]);
+            const { session, verifySurveyReadyEvent } = createSessionWithEventHandler(host);
+            const service = session.getProjectService();
+            openFilesForSession([file], session);
+            checkNumberOfProjects(service, { configuredProjects: 1 });
+            const project = service.configuredProjects.get(tsconfig.path)!;
+            checkProjectActualFiles(project, [file.path, tsconfig.path]);
+
+            verifySurveyReadyEvent();

```

Comment:
Few more tests:
- When tsconfig doesnt contain checkJs
- When project is closed and recreated, the event is not sent again.
- when tsconfig file without checkJS, project closed, updated the tsconfig to contain checkJs and reloaded -> Do you expect event in this scenario?